src/routes/courseRoutes.ts

```diff
@@ -0,0 +1,25 @@
+import { Router } from 'express';
+import { validate, validateParams } from '../middleware/validate';
+import {
+  createCourseValidator,
+  updateCourseValidator,
+  courseIdParamValidator
+} from '../validation';
+import {
+  createCourse,
+  updateCourse,
+  deleteCourse,
+  getCourse
+} from '../controllers/courseController';
+
+const router = Router();
+
+router.post('/courses', validate(createCourseValidator), createCourse);
+
+router.put('/courses/:id', validateParams(courseIdParamValidator), validate(updateCourseValidator), updateCourse);
```

Comment:
<img width="654" alt="image" src="https://github.com/user-attachments/assets/939192d5-cc73-4835-ae16-60e9487739bb">
the routes that use the validation should be like so...
route > validator > middleware > controller